### PR TITLE
ES6 crosswords grid

### PR DESCRIPTION
**Pull request by philwills**
## What does this change?
✨ ES6 ✨ the crossword's grid module

## What is the value of this and can you measure success?
Increased comprehensibility of this module

## Does this affect other platforms - Amp, Apps, etc?
No

## Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

No

## Screenshots
image

## Tested in CODE?
No

### CODE REVIEW

File: static/src/javascripts-legacy/projects/common/modules/crosswords/helpers.js [hunk 1]
```diff
@@ -291,7 +291,7 @@ define([
                 map[d.key] = {};
             }
 
-            map[d.key][d.direction] = d.separator;
+            map[d.key] = d;
```
Comment (philwills) [inline static/src/javascripts-legacy/projects/common/modules/crosswords/helpers.js]: This remapping complicated things in the grid module and I couldn't find it being used elsewhere.

File: static/src/javascripts/projects/common/modules/crosswords/grid.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+    render(): Element {
+        const width = helpers.gridSize(this.props.columns);
+        const height = helpers.gridSize(this.props.rows);
+        const cells = [];
+        let separators = [];
+
+        const range = n => Array.from({ length: n }, (value, key) => key);
+
+        const self = this;
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/crosswords/grid.js]: I don't think we need to do this, as arrow functions propagate the context through the call stack

**Review by SiAdcock — APPROVED** [review]
Minor comment, otherwise 👍 👍 👍

File: static/src/javascripts/projects/common/modules/crosswords/grid.js (hunk 2 above)
Comment (philwills) [reply static/src/javascripts/projects/common/modules/crosswords/grid.js]: Ah, yea, sorry, that was left from when I was trying to deal with something else. Will remove.

**Pull request merged**